src/legacy/ui/public/index_patterns/_pattern_cache.js

```diff
@@ -17,32 +17,35 @@
  * under the License.
  */
 
-export function IndexPatternsPatternCacheProvider() {
+export function createIndexPatternCache() {
 
   const vals = {};
+  const cache = {};
```

Comment:
Nitpick - I found the naming of these two objects to be a bit confusing. Usually a `cache` is for storing data but here its providing functionality. I think the issue of naming this object can be sidestepped, just return it with its methods defined inline.